velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -807,6 +807,55 @@ struct MakeYMIntervalFunction {
   }
 };
 
+template <typename T>
+struct MakeDTIntervalFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(out_type<IntervalDayTime>& result) {
+    return call(result, 0, 0, 0, 0);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(out_type<IntervalDayTime>& result, const int32_t& days) {
+    return call(result, days, 0, 0, 0);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(out_type<IntervalDayTime>& result, const int32_t& days, const int32_t& hours) {
+    return call(result, days, hours, 0, 0);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(out_type<IntervalDayTime>& result, const int32_t& days, const int32_t& hours, const int32_t& minutes) {
+    return call(result, days, hours, minutes, 0);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<IntervalDayTime>& result,
+      const int32_t days,
+      const int32_t hours,
+      const int32_t minutes,
+      const double seconds) {
+
+    static constexpr int64_t MICROS_PER_SECOND = 1'000'000;
+    static constexpr int64_t MICROS_PER_MINUTE = MICROS_PER_SECOND * 60;
+    static constexpr int64_t MICROS_PER_HOUR = MICROS_PER_MINUTE * 60;
+    static constexpr int64_t MICROS_PER_DAY = MICROS_PER_HOUR * 24;
```

Comment:
Perhaps reuse the variables defined in struct Timestamp.

https://github.com/facebookincubator/velox/blob/d9f1694f9d0d56f5ab59aba141b594b2eca26ea8/velox/type/Timestamp.h#L84-L85